### PR TITLE
feat(site): display build logs history in the build log page

### LINKED ISSUE #7409: Show historical startup logs in build logs page
Currently, Coder only displays the startup logs from the last run, and only if the workspace is started. We had a prospect request the ability to see startup logs alongside the historical build logs in Coder.

Currently, logs are only retained for a week or so. That is fine.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Close https://github.com/coder/coder/issues/7409

Demo:

https://github.com/coder/coder/assets/3165839/61309ac3-ccf2-40a5-a15f-918d27286375

### CODE REVIEW

**Review by mafredri — APPROVED** [review]
Nice change!

A few things I noticed clicking around:

1. I kind of miss a "back to workspace" button, I know the workspace name is clickable, but it feels a bit hidden away / not natural flow
2. The `Builds` list could also show `#1`, `#2`, etc.
3. When opening the builds page, the `Builds` list quickly flashes the loading indicator, don't we already have the builds since they're listed in the workspace page? Could we re-use that for a smoother experience?

**Comment by BrunoQuaresma** [comment]
@mafredri very valid points!

1. This is a good one, I will see how I can solve this a next time. This is a problem that is related not only to this page but the full-page header layout. 
2. That is true, I just don't see how to add it without making it too heavy visually but I will think of something.
3. Even the loading indicator being a skeleton (an element placeholder) you found it strange? Right now, we are not keeping the previous data because we don't have a single way to fetch server data, I hope to fix that this year but it would happen in a page refresh for example.

**Comment by mafredri** [comment]
> 3. Even the loading indicator being a skeleton (an element placeholder) you found it strange? Right now, we are not keeping the previous data because we don't have a single way to fetch server data, I hope to fix that this year but it would happen in a page refresh for example.

Yeah, it felt a bit weird since it resulted in some flashing (as in, it appeared and disappeared quite quickly). I think in a full page reload it's fine, both because of expectations and it will likely not flash as much since loading might take longer (multiple requests).

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.stories.tsx [hunk 1]
```diff
@@ -1,31 +1,48 @@
...
+const defaultBuilds = [...Array(15).keys()].map((i) => ({
+  ...MockWorkspaceBuild,
+  id: `${i}`,
+  build_number: i,
+}))
 
-export default {
-  title: "pages/WorkspaceBuildPageView",
+const meta: Meta<typeof WorkspaceBuildPageView> = {
+  title: "components/WorkspaceBuildPageView",
```
Comment (aslilac) [inline site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.stories.tsx:16]: hmm? this component is in pages/

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx [hunk 2]
```diff
@@ -23,38 +37,238 @@ const sortLogsByCreatedAt = (logs: ProvisionerJobLog[]) => {
...
-      )}
-
-      <Stack spacing={4}>
-        {build &&
-          build.transition === "delete" &&
-          build.job.status === "failed" && (
-            <WorkspaceBuildStateError build={build} />
+    <Box
+      sx={{
+        height: "calc(100vh - 62px - 36px)",
```
Comment (aslilac) [inline site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx:59]: can we have a comment explaining these values?

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx [hunk 3]
```diff
@@ -23,38 +37,238 @@ const sortLogsByCreatedAt = (logs: ProvisionerJobLog[]) => {
...
+          overflow: "hidden",
+          flex: 1,
+        }}
+      >
+        <Sidebar>
+          <SidebarCaption>Builds</SidebarCaption>
+          {!builds &&
+            [...Array(15).keys()].map((i) => (
+              <BuildSidebarItemSkeleton key={i} />
+            ))}
```
Comment (aslilac) [inline site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx:128]: `Array.from({ length: 15 }, (_, i) => ...)` does the same thing but allocates less and is a bit nicer imo :)

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx [hunk 4]
```diff
@@ -23,38 +37,238 @@ const sortLogsByCreatedAt = (logs: ProvisionerJobLog[]) => {
...
+              </Box>
+            </Alert>
+          )}
+          {!logs && <Loader />}
+          {logs && (
+            <WorkspaceBuildLogs
+              sx={{ border: 0 }}
+              logs={sortLogsByCreatedAt(logs)}
+            />
           )}
```
Comment (aslilac) [inline site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx:174]: could be a ternary?

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.stories.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.stories.tsx:16]: Ahhhh good catch!

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx:174]: 100%

File: site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/pages/WorkspaceBuildPage/WorkspaceBuildPageView.tsx:128]: TIL

**Pull request merged**